aleph-client/src/rpc.rs

```diff
@@ -10,18 +11,95 @@ fn json_req(method: &str, params: Value, id: u32) -> Value {
     })
 }
 
-pub fn author_rotate_keys() -> Value {
+pub fn author_rotate_keys_json() -> Value {
     json_req("author_rotateKeys", Value::Null, 1)
 }
 
+fn state_query_storage_at_json(storage_keys: &Vec<StorageKey>) -> Value {
+    json_req(
+        "state_queryStorageAt",
+        Value::Array(vec![
+            Value::Array(
+                storage_keys
+                    .into_iter()
+                    .map(|storage_key| Value::String(hex::encode(storage_key)))
+                    .collect::<Vec<_>>(),
+            ),
+            Value::Null,
+        ]),
+        1,
+    )
+}
+
+fn parse_query_storage_at_result(
+    maybe_json_result: Option<String>,
+) -> Result<Vec<Option<String>>, String> {
+    match maybe_json_result {
```

Comment:
To be honest, I'm not a fan of this function's signature. Building on what @kostekIV mentioned, I'm going to add that it's a bit suspicious that it takes a single optional value and throws an error if it's `None`. This seems like something better handled on use-site, especially given the argument above about deferring the missing-value case to the caller.

Reply:
Right, but those are all private functions - nobody from this module will be calling them, so I'd say it's okay. I blindly extracted function as my IDE suggested :) I can rework it though if you'd like to

Reply:
No worries, it’s okay, just something that I found a bit strange-looking 🙂